wordpress/wp-content/plugins/memberful-wp/src/search_result_enhancements.php

```
@@ -0,0 +1,153 @@
+  // Validate post_id is numeric and exists
+  if ( ! $post_id || ! is_numeric( $post_id ) || ! get_post( $post_id ) ) {
+    return $excerpt;
+  }
+
+  // Check if this post is protected and user doesn't have access
+  if ( ! memberful_can_user_access_post( get_current_user_id(), $post_id ) ) {
+    // Check if disclaimer should be shown
+    $show_disclaimer = get_option( 'memberful_show_search_disclaimer', TRUE );
+    
+    if ( $show_disclaimer ) {
+      $link_destination = get_option( 'memberful_search_link_destination', 'post' );
```

PHPMD (2.15.0)

Avoid unused local variables such as '$link_destination'. (undefined)

(UnusedLocalVariable)